src/main/java/org/sagebionetworks/web/client/widget/entity/editor/APITableColumnManager.java

```diff
@@ -15,63 +15,169 @@
+	public APITableColumnManager(APITableColumnManagerView view, PortalGinInjector ginInjector) {
 		this.view = view;
+		this.ginInjector = ginInjector;
 		view.setPresenter(this);
+		selectionChangedCallback = new Callback() {
+			@Override
+			public void invoke() {
+				checkSelectionState();
+			}
+		};
 	}
 	
 	@Override
 	public void configure(List<APITableColumnConfig> configs) {
-		this.configs = configs;
-		view.configure(configs);
+		columns = new ArrayList<APITableColumnConfigView>();
+		if (configs != null) {
+			for (APITableColumnConfig data : configs) {
+				APITableColumnConfigView column = ginInjector.getAPITableColumnConfigView();
+				column.setSelectionChangedCallback(selectionChangedCallback);
+				column.configure(data);
+				columns.add(column);
+			}
+		}
+		refreshColumns();
 	}
 
+
+	private void refreshColumns() {
+		view.clearColumns();
+		for (APITableColumnConfigView column : columns) {
+			view.addColumn(column.asWidget());
+		}
+		boolean columnsVisible = columns.size() > 0;
+		view.setButtonToolbarVisible(columnsVisible);
+		view.setHeaderColumnsVisible(columnsVisible);
+		view.setNoColumnsUIVisible(!columnsVisible);
+		checkSelectionState();
+	}
+	
 	@Override
 	public Widget asWidget() {
 		return view.asWidget();
 	}
 	
 	@Override
-	public void addColumnConfig(String rendererName, String inputColumnNames,
-			String displayColumnName, COLUMN_SORT_TYPE sort) {
-		if (rendererName == null || inputColumnNames == null || rendererName.trim().length() == 0 || inputColumnNames.trim().length() == 0) {
-			throw new IllegalArgumentException("Renderer and Input Columns are required");
-		}
-		APITableColumnConfig newConfig = new APITableColumnConfig();
-		newConfig.setRendererFriendlyName(rendererName);
-		String[] inputColNamesArray = inputColumnNames.split(",");
-		Set<String> inputColumnNamesSet = new HashSet<String>();
-		for (int i = 0; i < inputColNamesArray.length; i++) {
-			inputColumnNamesSet.add(inputColNamesArray[i].trim());
+	public void addColumnConfig() {
+		APITableColumnConfigView column = ginInjector.getAPITableColumnConfigView();
+		column.setSelectionChangedCallback(selectionChangedCallback);
+		column.configure(new APITableColumnConfig());
+		columns.add(column);
+		refreshColumns();
+	}
+	
+
+	public void selectAll() {
+		changeAllSelection(true);
+	}
+
+	public void selectNone() {
+		changeAllSelection(false);
+	}
+
+	public void onMoveUp() {
+		int index = findFirstSelected();
+		APITableColumnConfigView sourceEditor = columns.get(index);
+		columns.remove(index);
+		columns.add(index-1, sourceEditor);
+		refreshColumns();
+	}
+
+	public void onMoveDown() {
+		int index = findFirstSelected();
+		APITableColumnConfigView sourceEditor = columns.get(index);
+		columns.remove(index);
+		columns.add(index+1, sourceEditor);
+		refreshColumns();
+	}
+
+	public void deleteSelected() {
+		Iterator<APITableColumnConfigView> it = columns.iterator();
```

Comment:
This could be refactored so that all list objects extend the same interface, then the same selection toolbar could also have this list.